### PR TITLE
RAT-338: Bulk refresh for plugin

### PR DESCRIPTION
**Pull request by cstamas**
Refresh deps but also code. And please release, as current last release is a Maven 2 plugin:
https://deps.dev/maven/org.apache.rat%3Aapache-rat-plugin/0.15/dependencies

---

https://issues.apache.org/jira/browse/RAT-338

### CODE REVIEW

**Comment by ottlinger** [comment]
@cstamas thanks for your contributions - we are about to push out a new release, but work is still going on concerning some performance regressions apart from moving to Maven3.
Pls rebase your PR as some of the dependabot updates were merged in.

Can you be so kind to create a https://issues.apache.org/jira/browse/RAT an reference it in your PR? Thanks again for your help!

File: pom.xml [hunk 1]
```diff
@@ -52,12 +52,13 @@ agnostic home for software distribution comprehension and audit tools.
     <maven.compiler.source>${javaVersion}</maven.compiler.source>
     <maven.compiler.target>${javaVersion}</maven.compiler.target>
     <!-- This is the version of Maven required to use the Rat Maven Plugin -->
-    <mavenVersion>3.2.5</mavenVersion>
+    <mavenMinVersion>3.2.5</mavenMinVersion>
+    <mavenVersion>3.9.6</mavenVersion>
```
Comment (ottlinger) [inline pom.xml:56]: fyi: I started a discussion on the mailing list concerning this update ...... let's see if there are any complaints about that one.

Comment (cstamas) [reply pom.xml:56]: The point of this update: "we" (Maven project) did it wrong, compile against latest (3.9.6) but allow runtime 3.2.5 (min maven). The reason to do so is to pick up deprecations and get notified what is changing, but source and binary compatibility guarantees it to run on min version as well. If you compile against 3.2.5 forever, you get totally disconnected about latest changes and deprecations in Maven APIs.

File: apache-rat-plugin/src/main/java/org/apache/rat/mp/AbstractRatMojo.java [hunk 2]
```diff
@@ -392,7 +392,7 @@ private void setIncludes(DirectoryScanner ds) throws MojoExecutionException {
                 }
                 includeList.addAll(getPatternsFromFile(f, charset));
             }
-            ds.setIncludes(includeList.toArray(new String[includeList.size()]));
+            ds.setIncludes(includeList.toArray(new String[0]));
```
Comment (ottlinger) [inline apache-rat-plugin/src/main/java/org/apache/rat/mp/AbstractRatMojo.java:395]: Following https://docs.oracle.com/javase/8/docs/api/java/util/Collection.html#toArray-T:A- setting the array's size prevents creating a new one.

File: pom.xml (hunk 1 above)
Comment (cstamas) [reply pom.xml:56]: Also, this affects _compilation only_, as at runtime, whatever Maven version you use, will provide the runtime stuff, and this does not mean Maven 3.9.6 is must or needed or whatever. Again, this is only to get compiler emit deprecation messages to notify devs (like use of deprecated type ArtifactRepository in one mojo is).

Comment (ottlinger) [reply pom.xml:56]: Thanks for this context. So basically it's the same as https://github.com/apache/creadur-rat/pull/172 concerning the maven deps?

Comment (cstamas) [reply pom.xml:56]: Not quite: that is "machine" going for highest, for example it left unreal "min" version in place:
https://github.com/apache/creadur-rat/blob/b0442fd49026b1929e9a88f2ea1c655dca106115/pom.xml#L469

Maven 3.0.x should be avoided (due package rename happened between 3.0.x and 3.1.x). We in Maven also set "lowest version" to 3.2.5 these days, it covers from 2014 until today, I hope 10 years is good enough.

**Comment by ottlinger** [comment]
@cstamas thanks for your contributions, fixes and explanations.

**Pull request merged**